app/main.py

```diff
@@ -1,2 +1,48 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 5
+        self.filled_cells = 0
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(self,
+                    key: int | float | str | bool | tuple,
+                    value: Any) -> None:
+        index = hash(key) % self.capacity
+        if self.hash_table[index] is None:
+            self.hash_table[index] = []
+        for cell in self.hash_table[index]:
+            if cell[0] == key:
+                cell[1] = hash(key)
+                cell[2] = value
+                return
+        self.hash_table[index].append([key, hash(key), value])
+        self.filled_cells += 1
```

Comment:
Are you sure the collisions will be handled properly? The index should be incremented when the collission occurrs. Here the index remains unchanged